### PR TITLE
Fix build app script

### PR DESCRIPTION
**Pull request by Ruben-Crispino**
Improved build_app script to manage multiple backups.

### CODE REVIEW

File: build_app.sh [hunk 1]
```diff
@@ -10,12 +10,23 @@ application="Applications/$1"
...
+if [ "$(ls -A app_src -I ".gitkeep")" ]
+then
+  mv app_src/* $BACKUP_DIR/$NEW_DIR_NAME
+  cp -r $application/. app_src/
+else
+   cp -r $application/. app_src/
+fi
+
+#SUDO privileges are required
+sudo ./docker_build_image.sh
```
Comment (magiczny-kacper) [inline build_app.sh:32]: Why? You don't and should not need sudo privileges to build application.

**Review by magiczny-kacper — CHANGES_REQUESTED** [review]

File: build_app.sh (hunk 1 above)
Comment (Ruben-Crispino) [reply build_app.sh:32]: Docker needs a sudo user to run as described here: https://docs.docker.com/engine/install/linux-postinstall/
If this needs to be avoided the developer need to create a new group and add himeself to that group. 
Is there any specific reason why sudo privileges needs to be avoided?

Comment (magiczny-kacper) [reply build_app.sh:32]: Using sudo in build script complicates the building process in my opinion. Adding docker to new group is needed just once, and then, you can build app with few key strokes. Using sudo inside scripts require sthe user to enter a password each time, which, in my opinion, sometimes might be annoying, especially, when for example, you just miss spelled the variable name and need to build just after the previous build.

Comment (Ruben-Crispino) [reply build_app.sh:32]: It is perfectly fine to me given that this is described somewere in the documentations. Not all developers are familiar with linux and hence these details can be ignored raising issues.
What about managing the group creation automatically on the script? Would that be something usefull in your opinion?

Comment (magiczny-kacper) [reply build_app.sh:32]: @Ruben-Crispino yes, it would be useful in my opinion

Comment (Ruben-Crispino) [reply build_app.sh]: @magiczny-kacper check new commit

**Review by magiczny-kacper — APPROVED** [review]

**Pull request merged**